### PR TITLE
Update to go 1.22

### PR DESCRIPTION
**Pull request by lcarva**
Ref: EC-598

### CODE REVIEW

**Review by robnester-rh — APPROVED** [review]

File: .tool-versions [hunk 1]
```diff
@@ -1 +1 @@
-golang 1.21.9
+golang 1.22.5
```
Comment (zregvart) [inline .tool-versions:1]: ~~1.22.6 is newer, perhaps use that?~~ I see the `brew.registry.redhat.io/rh-osbs/openshift-golang-builder:v1.22` is at `go version go1.22.5 (Red Hat 1.22.5-1.el9) linux/amd64`, makes sense to be on the same patch version.

I guess we could enforce that all versions used here are matched in CI somehow? Would that be something of value?

File: Dockerfile [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 ## Build
 
-FROM docker.io/library/golang:1.21 AS build
+FROM docker.io/library/golang:1.22 AS build
```
Comment (zregvart) [inline Dockerfile:19]: Not sure if we want to, but we could pin to the patch version here...

```suggestion
FROM docker.io/library/golang:1.22.5 AS build
```

File: Dockerfile.dist [hunk 3]
```diff
@@ -17,9 +17,11 @@
 ## Build
 
 # This works fine but will produce an EC violation
-# FROM docker.io/library/golang:1.21 AS build
+# FROM docker.io/library/golang:1.22 AS build
```
Comment (zregvart) [inline Dockerfile.dist:20]: Do we need these comments, perhaps its a good opportunity to remove them?

File: Dockerfile.dist [hunk 4]
```diff
@@ -17,9 +17,11 @@
...
 
-FROM registry.access.redhat.com/ubi9/go-toolset:1.21@sha256:ae17d73e70a966f39ef4dfca74241e3ca4374cd1198b02c30ea0748b8dcc83a6 AS build
+# Ideally, use the official image from Red Hat, but a 1.22 release does not yet exist.
+# FROM registry.access.redhat.com/ubi9/go-toolset:1.21@sha256:ae17d73e70a966f39ef4dfca74241e3ca4374cd1198b02c30ea0748b8dcc83a6 AS build
```
Comment (zregvart) [inline Dockerfile.dist:23]: I'd remove this it is pointing to 1.21...

```suggestion
```

**Review by zregvart — APPROVED** [review]
:rocket:

File: Dockerfile (hunk 2 above)
Comment (lcarva) [reply Dockerfile:19]: I think it's ok to just use `1.22`. Go is usually pretty stable between patch releases.

File: .tool-versions (hunk 1 above)
Comment (lcarva) [reply .tool-versions:1]: Yeah. Not sure if we can build the project if the version on the builder image is older (even just at the patch version). Need to play with it further.

File: Dockerfile.dist (hunk 4 above)
Comment (lcarva) [reply Dockerfile.dist]: Fair. I re-wrote the comment and remove that specific image ref.

File: Dockerfile.dist (hunk 3 above)
Comment (lcarva) [reply Dockerfile.dist]: Good idea.

File: .tool-versions (hunk 1 above)
Comment (lcarva) [reply .tool-versions:1]: Following up in https://github.com/enterprise-contract/ec-cli/pull/1917.

> I guess we could enforce that all versions used here are matched in CI somehow? Would that be something of value?

Maybe let's see if not doing that causes any issues?

**Pull request merged**